src/main/java/com/almostreliable/merequester/requester/status/PlanState.java

```diff
@@ -11,12 +12,33 @@
 
 public final class PlanState implements StatusState {
 
+    private final boolean wasMissing;
     private final Future<? extends ICraftingPlan> future;
 
-    PlanState(Future<? extends ICraftingPlan> future) {
+    PlanState(boolean wasMissing, Future<? extends ICraftingPlan> future) {
+        this.wasMissing = wasMissing;
         this.future = future;
     }
 
+    static PlanState BuildPlan(RequesterBlockEntity host, int slot, boolean wasMissing)
```

Comment:
This is completely different code style from the rest of the project. Methods should always be lowercase even if they are static. You also should not place the opening bracket in the next line.

Reply:
Sorry, C# bleeding through and I apparently hate myself because I setup vscode for Java right as java22 broke a bunch of extensions. I should have double checked formatting though, that's on me.